src/ape/managers/_contractscache.py

```diff
@@ -577,15 +577,21 @@ def get(
                     if fetch_from_explorer
                     else None
                 )
-                if proxy_contract_type:
-                    contract_type_to_cache = _get_combined_contract_type(
+                if proxy_contract_type is not None and implementation_contract_type is not None:
+                    combined_contract = _get_combined_contract_type(
                         proxy_contract_type, proxy_info, implementation_contract_type
                     )
-                else:
+                    self.contract_types[address_key] = combined_contract
+                    return combined_contract
+
+                elif implementation_contract_type is not None:
                     contract_type_to_cache = implementation_contract_type
+                    self.contract_types[address_key] = implementation_contract_type
+                    return contract_type_to_cache
 
-                self.contract_types[address_key] = contract_type_to_cache
-                return contract_type_to_cache
+                elif proxy_contract_type is not None:
```

Comment:
so actually, this scenario is kind of a bug because if we know it's a proxy but don't know it's implementation ABI, you can't really do much with it

Reply:
You can have a proxy without it having an implementation set yet, say for example you deployed it yourself but have yet to set the implementation. I don't see how that is a bug.

Reply:
There are still ABIs available on the proxy contract itself, still things you can do etc.

Reply:
> You can have a proxy without it having an implementation set yet, say for example you deployed it yourself but have yet to set the implementation. I don't see how that is a bug.

that's actually a good point, okay I'm on board now

Reply:
Oh wait, no you're right... In this case, the implementation contract _is_ set, but for some reason, we were unable to fetch its contact type or we don't know it. 🤔

Reply:
I suppose it could be the case that the user isn't using an explorer and purposely deleted the contract type from the cache, which is a valid set of moves.

Reply:
It gets confusing though, because it is even hard to know if an implementation is set or not, as it may be a placeholder address